### PR TITLE
Make post-build.yml accept parameters to sdk-task.ps1

### PR DESCRIPTION
**Pull request by JohnTortugo**
Closes: #3666  #3661
Tests: ~~[310059](https://dnceng.visualstudio.com/internal/_build/results?buildId=310059&view=results)~~ [310150](https://dnceng.visualstudio.com/internal/_build/results?buildId=310150&view=results)

Add new parameters to post-build.yml that can be used to:

- Pass parameters to sdk-tasks.ps1 which let users to customize the invocation of sdk-tasks.
- Set default values for DotNetArtifactsCategory and DotNetValidationArtifactsCategory

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -19,10 +19,6 @@ variables:
     value: DotNetCore
   - name: _PublishUsingPipelines
     value: true
-  - name: _DotNetArtifactsCategory
```
Comment (riarenas) [inline azure-pipelines.yml]: I'd still like these variables here so people who look at Arcade's YAML for examples can see how to define them. Do you feel strongly about removing them?

Comment (JohnTortugo) [reply azure-pipelines.yml]: I'm totally fine with that

File: eng/common/templates/post-build/post-build.yml [hunk 2]
```diff
@@ -7,6 +7,16 @@ parameters:
     enable: false
     params: ''
 
+  # These parameters let the user customize the call to sdk-task.ps1 for publishing
+  # symbols & general artifacts as well as for signing validation
+  symbolPublishingAdditionalParameters: ''
+  artifactsPublishingAdditionalParameters: ''
+  signingValidationAdditionalParams: ''
```
Comment (riarenas) [inline eng/common/templates/post-build/post-build.yml]: [nit] We use the full word `parameters` everywhere else.

Comment (JohnTortugo) [reply eng/common/templates/post-build/post-build.yml]: Good, thanks Ricardo

**Pull request merged**